ush/om_data.py

```diff
@@ -0,0 +1,390 @@
+        """
+        Initialize OM_data object and copy data to requested location.
+        """
+
+        self.pdate = datetime.strftime(config.get('PDATE'), '%Y%m%d%H')
+        self.data_src = data_src
+        self.logger = logger
+        self.data_dir_type = ""
+        self.plot_dict = {}
+        self.ctl_file = []
+        self.data_files = []
+        self.dir_struct = ""
+        self.data_type = ""
+        self.data_subtype = ""
+
+        self.load_data_files(plot_yaml)
+        self.set_data_types(os.path.basename(config.get('PLOT_TEMPLATE')))
+        self.set_dir_struct()
+
+        self.dump()
+        self.copy_data(os.getcwd())
+
+# --------------------------------------------------------------------------------------------
+
+    def dump(self):
+        """
+        Dump all current values in object of class OM_data via logger.info.
+
+        Paremeters:
+            None
+        Return:
+            None
+        """
+
+        self.logger.info(f'\n' + 'dump OM_data:')
+        self.logger.info(f'============')
+        self.logger.info(f'    data_type: {self.data_type}')
+        self.logger.info(f' data_subtype: {self.data_subtype}')
+        self.logger.info(f'        pdate: {self.pdate}')
+        self.logger.info(f'     data_src: {self.data_src}')
+        self.logger.info(f'     ctl_file: {self.ctl_file}')
+        self.logger.info(f'   data_files: {self.data_files}')
+        self.logger.info(f'   dir_struct: {self.dir_struct}')
+        self.logger.info('============' + '\n')
+
+# --------------------------------------------------------------------------------------------
+
+    def set_data_types(self, template):
+        """
+        Set data_type and data_subtype using the plot template name.  Note
+        that 'con' and 'ozn' types have some special cases for subtype.
+
+        Paremeters:
+            template (str): Name of plot template file
+        Return:
+            None
+        """
+
+        self.data_type = template[:3].lower()
+        stype = template[3:].lower()
+
+        self.data_subtype = stype
+
+        match self.data_type:
+            case 'con':
+                if stype in ['time', 'vert']:
+                    self.data_subtype = 'time_vert'
+                elif stype in ['hist', 'horz']:
+                    self.data_subtype = 'horz_hist'
+
+            case 'ozn':
+                if stype == 'summary':
+                    self.data_subtype = 'time'
+
+# --------------------------------------------------------------------------------------------
+
+    def load_data_files(self, yaml_file):
+
+        """
+        Load control and data file names from plot_dict into OM_data class object.
+
+        Parameters:
+            yaml_file (file): yaml file with plot dict info
+        Return:
+            None
+        """
+
+        try:
+            with open(yaml_file, 'r') as yaml_file_opened:
+                pd = yaml.safe_load(yaml_file_opened)
```

Comment:
can we use `wxflow` tools so that we can do Jinja templating down the road?

Reply:
I will investigate that.  I confess I'm largely ignorant of `wkflow`'s capabilities.

Reply:
Yes, we can use `wxflow`'s `parse_yaml()` to read the yaml file.